Source/Scene/Cesium3DTileGroundPrimitive.js

```diff
@@ -0,0 +1,875 @@
+        var batchedIds = new Uint16Array(positionsLength / 3 * 2);
+        var batchedIndexOffsets = new Array(indexOffsets.length);
+        var batchedIndexCounts = new Array(indexCounts.length);
+
+        // TODO: compute length and create typed array
+        var batchedIndices = [];
+
+        var colors = primitive._colors;
+        var colorsLength = colors.length;
+
+        var i;
+        var j;
+        var color;
+        var rgba;
+
+        var buffers = {};
+        for (i = 0; i < colorsLength; ++i) {
+            rgba = colors[i].toRgba();
+            if (!defined(buffers[rgba])) {
+                buffers[rgba] = {
+                    positionLength : counts[i],
+                    indexLength : indexCounts[i],
+                    offset : 0,
+                    indexOffset : 0,
+                    batchIds : [i]
+                };
+            } else {
+                buffers[rgba].positionLength += counts[i];
+                buffers[rgba].indexLength += indexCounts[i];
+                buffers[rgba].batchIds.push(i);
+            }
+        }
+
+        var object;
+        var byColorPositionOffset = 0;
+        var byColorIndexOffset = 0;
+        for (rgba in buffers) {
+            if (buffers.hasOwnProperty(rgba)) {
+                object = buffers[rgba];
+                object.offset = byColorPositionOffset;
+                object.indexOffset = byColorIndexOffset;
+
+                var positionLength = object.positionLength * 2;
+                var indexLength = object.indexLength * 2 + object.positionLength * 6;
+
+                byColorPositionOffset += positionLength;
+                byColorIndexOffset += indexLength;
+
+                object.indexLength = indexLength;
+            }
+        }
+
+        var batchedDrawCalls = [];
+
+        for (rgba in buffers) {
+            if (buffers.hasOwnProperty(rgba)) {
+                object = buffers[rgba];
+
+                batchedDrawCalls.push({
+                    color : Color.fromRgba(parseInt(rgba)),
+                    offset : object.indexOffset,
+                    count : object.indexLength,
+                    batchIds : object.batchIds
+                });
+            }
+        }
+
+        primitive._batchedIndices = batchedDrawCalls;
+
+        for (i = 0; i < colorsLength; ++i) {
+            color = colors[i];
+            rgba = color.toRgba();
+
+            object = buffers[rgba];
+            var positionOffset = object.offset;
+            var positionIndex = positionOffset * 3;
+            var colorIndex = positionOffset * 4;
+            var idIndex = positionOffset;
+
+            var polygonOffset = offsets[i];
+            var polygonCount = counts[i];
+
+            var minLat = Number.POSITIVE_INFINITY;
+            var maxLat = Number.NEGATIVE_INFINITY;
+            var minLon = Number.POSITIVE_INFINITY;
+            var maxLon = Number.NEGATIVE_INFINITY;
+
+            for (j = 0; j < polygonCount; ++j) {
```

Comment:
Is this loop a hot spot when profiling load?  What was the plan for web workers here?  Doesn't need to be in this PR, but should it be in the near-term roadmap?